### PR TITLE
FTP working + added binary mode + removed preliminary projects

### PR DESCRIPTION
**Pull request by sgtz**
added use binary mode param and defaulted to this
added various intellisense comments
removed diagnostic -> stdout lines used for debugging
tidied up a few things

Not sure if it's okay to remove the preliminary projects... I was getting this cleaned up for FSharp.Management

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,12 @@
+FtpProvider/FtpProvider/bin/Debug/FSharp.Core.dll
```
Comment (dsyme) [inline .gitignore]: FWIW you can just ignore whole directories

File: FtpProvider/FtpProvider/FtpProvider.fs [hunk 2]
```diff
@@ -1,102 +1,161 @@
...
+    let dirs = 
+        [ for c in contents do 
+            if c.Length > 1 then 
+               if c.[0].StartsWith("d") then yield Seq.last c ]
+
+    let files = 
+        [ for c in contents do 
+            if c.Length > 1 then 
+               if c.[0].StartsWith("-") then yield Seq.last c ]
+
```
Comment (dsyme) [inline FtpProvider/FtpProvider/FtpProvider.fs:36]: The technique used above parses the results of "list directory".  It would be good to check if different FTP servers give directory listings in compatible textual formats

File: FtpProvider/FtpProvider/FtpProvider.fs [hunk 3]
```diff
@@ -1,102 +1,161 @@
...
+    inherit TypeProviderForNamespaces ()
+    let nameSpace = "FSharp.Management"
+    let asm = Assembly.GetExecutingAssembly()
+
+    // Recursive, on-demand adding of types
+    let createTypes (typeName, site, user, pwd:string, useBinary:bool) = 
+        let rec addTypes (site:string, td:ProvidedTypeDefinition) =
+
+            td.AddMembersDelayed(fun () -> 
+                // TODO.6
```
Comment (dsyme) [inline FtpProvider/FtpProvider/FtpProvider.fs]: What's this TODO for?

File: FtpProvider/FtpProvider/FtpProvider.fs [hunk 4]
```diff
@@ -1,102 +1,161 @@
...
+        actualType
+
+    let addProvidedStaticParameter nme typ xmldoc = 
+      let p = ProvidedStaticParameter(nme,typ) 
+      p.AddXmlDoc(sprintf xmldoc)
+      p
+
+    let _ = 
+        let a = ProvidedTypeDefinition(asm, nameSpace, "FtpProvider", Some typeof<obj>)
+        a.AddXmlDoc("An FTP Type Provider which lets you 'dot' into directory structures, and then retrieve a file by 'dotting' into the '.Contents' tag.  note: there are no progress updates, so if it's a large file over a slow connection, the only solution is to wait.  Perhaps try a smaller file first to verify.")  // BUG.2
```
Comment (dsyme) [inline FtpProvider/FtpProvider/FtpProvider.fs]: It looks like type `a` is never used.  it should be deleted and the XmlDoc attached to `topType`?

File: FtpProvider/FtpProvider/FtpProvider.fs [hunk 5]
```diff
@@ -1,102 +1,161 @@
...
+
+// DONE
+// ----
+// TODO.7: get back files in various formats such as text and binary.  Include means to be able to flick on / off all the important FTP switches
+
+// FIXED
+// -----
+// BUG.1: file identifiers are never defined.  Is this because of an error in the quotation?  Why?
+
+// RESOLVED
```
Comment (dsyme) [inline FtpProvider/FtpProvider/FtpProvider.fs]: Remove these notes for resolved/fixed issues

File: FtpProvider/FtpProvider/FtpProvider.fs [hunk 6]
```diff
@@ -1,102 +1,161 @@
...
+        this.AddNamespace(nameSpace, [topType])
+
+[<assembly:TypeProviderAssembly>]
+do ()
+
+// TODO
+// ----
+
+// TODO.1: pick up FTP courtesy details from environment variables?
+// TODO.2: add progress updates via intellisense?  ie. |          |  and |****  underneath for a text based left to right gauge
```
Comment (dsyme) [inline FtpProvider/FtpProvider/FtpProvider.fs:186]: Asynchronous loading like this could indeed be useful but probably more so at runtime (loading big files) rather than at compile-time (getting directory contents). Adding a `GetContentsAsync()` method (and changing `Contents` property to a `GetContents()` method for consistency) would be one way to do this.

File: FtpProvider/FtpProvider/FtpTypeProvider.fsproj [hunk 7]
```diff
@@ -51,7 +51,6 @@
   <ItemGroup>
     <Compile Include="ProvidedTypes.fsi" />
     <Compile Include="ProvidedTypes.fs" />
-    <Content Include="packages.config" />
```
Comment (dsyme) [inline FtpProvider/FtpProvider/FtpTypeProvider.fsproj:54]: Why is this removed?

**Comment by dsyme** [comment]
This is looking good. See various comments above, and I'll merge once those are done and then you can move everything across to FSharp.Management?

File: FtpProvider/FtpProvider/FtpTypeProvider.fsproj (hunk 7 above)
Comment (sgtz) [inline FtpProvider/FtpProvider/FtpTypeProvider.fsproj:54]: > In FtpProvider/FtpProvider/FtpTypeProvider.fsproj
> https://github.com/dsyme/FtpTypeProviderExample/pull/5#discussion_r38634948
> :
> 
> > @@ -51,7 +51,6 @@
> >    <ItemGroup>
> >      <Compile Include="ProvidedTypes.fsi" />
> >      <Compile Include="ProvidedTypes.fs" />
> > -    <Content Include="packages.config"
> 
> Why is this removed?
> 
> Not removed.  I merged the two <ItemGroup> nodes for .dlls and source code
> into one.

The VS IDE stopped working correctly with the click-reference expand
behaviour.  I initially thought that the cause might be a HD corruption or
a  change in the .fsproj file.  I was looking for the cause + did a little
editing while there.   The cause was actually a VS plug-in -- probably the
GitHub one, because the most recent update fixed the issue.  I'll put it
back to the way it was.

Comment (sgtz) [inline FtpProvider/FtpProvider/FtpTypeProvider.fsproj:54]: "What's this TODO for?"

when comments get in the way of just reading the code, I came up with an
annotation convention.  i.e. TODO.6 refers a specific TODO item below.  In
my minds eye I imagined a tooling extension in VS for this.  Even in a raw
form I've found this useful though.  Since just a personal style, I'll
remove them.

Comment (sgtz) [inline FtpProvider/FtpProvider/FtpTypeProvider.fsproj:54]: "FWIW you can just ignore whole directories"

I figured.  That's me getting used to SourceTree from Altlassian

File: FtpProvider/FtpProvider/FtpProvider.fs (hunk 2 above)
Comment (sgtz) [inline FtpProvider/FtpProvider/FtpProvider.fs:36]: On 3 September 2015 at 21:12, Don Syme notifications@github.com wrote:

> The technique used above parses the results of "list directory". It would
> be good to check if different FTP servers give directory listings in
> compatible textual formats

Any examples of this that you can lay your hands on easily?  I had a hunt
around on google but didn't find anything.
